installation_and_upgrade/ibex_install_utils/tasks/backup_tasks.py

```diff
@@ -43,35 +43,36 @@ def zip_file(self, filename, zipname):
                     zipf.write(os.path.join(root, file), arcname=os.path.join(root, file).replace(filename, ''))
                     self.update_progress_bar(i, len(all_files))
                     i = i + 1
+                    
+    def move_file(self, src, dst, copy=False, ignore=None):
+        number_of_files = 0
+        current_file_index = 0
 
-    def move_file(self, src, dst, copy=False):
-        all_files = []
-        for root, dirs, files in os.walk(src):
-            for file in files:
-                all_files.append((os.path.join(root, file), os.path.join(dst, os.path.relpath(root, src))))
+        def count_files_copy_function(src, dst):
+            nonlocal number_of_files
+            number_of_files += 1
 
-        total_files = len(all_files)
+        def copy_function(src, dst):
+            nonlocal current_file_index
 
-        if copy:
-            operation = shutil.copy
-        else:
-            operation = shutil.move
+            if copy:
+                operation = shutil.copy
```

Comment:
The default `shutil.copytree` uses `shutil.copy2` to copy the file and also preserve file metadata, it may be useful to keep metadata ourselves too

Reply:
Were you referring to the symbolic links and incompatibility? I think that is mostly an issue with follow_symlinks=False   what i was most interested in keeping with `copy2()` was file modification dates as they may help with future diagnostics. It claims it never throws an error from a failure to copy metadata, so should not cause any problems trying to keep it.